retry/retry.go

```diff
@@ -81,3 +86,11 @@ func (r *Retry) waitTime(attempt uint) int {
 	total := (baseTimeSeconds + delay) * float64(time.Second)
 	return int(total)
 }
+
+// Test version, returns retry in test mode (nanosecond retry delay).
+func NewTestRetry(maxRetry uint) Retry {
```

Comment:
good idea! (note to self to use this in the future)

Reply:
I considered a 'Retrier' interface that I could swap out with a test version but the semantics of the interface meant the nil value resulted in a panic and the changes to fix that were ugly. So I preserved using the struct directly and added a test mode to it.